snapcraft/providers.py

```diff
@@ -197,15 +197,20 @@ def get_base_configuration(
 
     return bases.BuilddBase(
         alias=alias,
-        compatibility_tag=f"snapcraft-{bases.BuilddBase.compatibility_tag}.0",
+        compatibility_tag=f"snapcraft-{bases.BuilddBase.compatibility_tag}.1",
         environment=environment,
         hostname=instance_name,
         snaps=[
+            Snap(
```

Comment:
Should I make a new issue to only do this when `components` are defined?  
Currently, it is a subtask of https://github.com/canonical/snapcraft/issues/4644

It should be easy to implement (import craft_parts.features and check if partitions are enabled)

Reply:
I thought we would only merge here and hook up in a different task, so it seems appropriate

Reply:
Created: https://github.com/canonical/snapcraft/issues/4656